File: ManejoArchivosJava/src/ManejoArchivosJava/Moto.java
/*
 esta es la clase Moto que hereda de la clase vehiculo y ademas tambien 
posee los metodos de la clase cualidades
 */
package ManejoArchivosJava;

import ManejoArchivosJava.Vehiculo;
import javax.swing.JOptionPane;

/**
 * 28 septiembre 2018
 * @author Manuela García Monsalve 
 */
public class Moto extends Vehiculo implements Cualidades{
    
    String medio;
    String rueda;
    
    //constructor de la clase Moto
    
    public Moto(String marca, String modelo, String color, String medio,String rueda)
    {
        super(marca,modelo,color);
        this.medio = medio;
        this.rueda = rueda;
    }
     //sobrescritura de  los metodo de la interface cualidades
    @Override
    public void Prender()
    {
        JOptionPane.showMessageDialog(null, "me he prendido ");
    }
    @Override
    public void Arrancar()
    {
        JOptionPane.showMessageDialog(null, "he arrancado ");
    }
    @Override
    public void Apagar()
    {
        JOptionPane.showMessageDialog(null, "me he apagado");
    }
    
    public void Picar()
    {
        Prender();
        Arrancar();
        JOptionPane.showMessageDialog(null, "me puedo picar");
        Apagar();
    }
    
    @Override
    public void Reversa()
    {
        JOptionPane.showMessageDialog(null, "No puede dar reversa");
    }
    
        @Override
    public String VerDatos()
    {
        return  "Marca: " + this.marca 
            + "\nModelo: " + this.modelo + "\nColor: " + this.color
            + "\nMedio: " + this.medio;
    }
}

File: ManejoArchivos/Moto.py
#Autor Manuela Garcia Monsalve
# 28 septiembre 2018
#Subclase Moto que aparte de heredar de la super clase vehiculo, tambien tendra metodos propios
#como picar

from Vehiculo import *

class Moto(Vehiculo):

    def __init__(self,marca,color,modelo,medio,rueda):
        Vehiculo.__init__(marca,color,modelo)
        self.medio = medio
        self.rueda = rueda

    def Preder(self):
        pass

    def Apagar(self):
        pass
    def Arrancar(self):
        pass

    def Picar(self): #Metodo propio de la clase
        pass

